### PR TITLE
Add support for subscribing to config entry changes

### PR DESCRIPTION
**Pull request by bdraco**
## Proposed change

Needs https://github.com/home-assistant/core/pull/77803

The dispatcher is used to get events to the api since config entry state changes
are frequent and we do not want to broadcast them to the event bus.

I didn't use a collection here since config entries are rather noisy with retries
and state changes and I wanted to send everything in a single api so
there wasn't a race between fetching the initial payload and getting callbacks.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/config/integrations/ha-config-integrations.ts [hunk 1]
```diff
@@ -255,9 +261,81 @@ class HaConfigIntegrations extends SubscribeMixin(LitElement) {
...
+    super.connectedCallback();
+    if (this.hasUpdated) {
+      this._subscribeConfigEntries();
+    }
+  }
+
+  public disconnectedCallback() {
+    super.disconnectedCallback();
+    this._unsubscribeConfigEntries();
+  }
```
Comment (bramkragten) [inline src/panels/config/integrations/ha-config-integrations.ts:275]: Use the `SubscribeMixin` to handle this for you

File: src/panels/config/integrations/ha-config-integrations.ts [hunk 2]
```diff
@@ -255,9 +261,81 @@ class HaConfigIntegrations extends SubscribeMixin(LitElement) {
...
+
+  private _subscribeConfigEntries() {
+    this._subscribedConfigEntries = subscribeConfigEntries(
+      this.hass,
+      (messages) => {
+        let fullUpdate = false;
+        const newEntries: ConfigEntryExtended[] = [];
+        messages.forEach((message) => {
+          if (message.type === null || message.type === "added") {
```
Comment (bramkragten) [inline src/panels/config/integrations/ha-config-integrations.ts:284]: So the first response will be all integrations with type = `null`?

Comment (bdraco) [reply src/panels/config/integrations/ha-config-integrations.ts:284]: Yes. Current state is a non event so null

Comment (bdraco) [reply src/panels/config/integrations/ha-config-integrations.ts]: added a comment in 6a5e3d94ae42bc2f40921bf316d08d8c8fc98467

**Review by bramkragten — APPROVED** [review]

**Pull request merged**